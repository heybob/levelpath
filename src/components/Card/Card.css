.card {
    display: flex;
    width: 50%;
    border: 1px solid lightgray;
    border-radius: 1rem;
    box-shadow: 1px 1px 10px rgb(0,0,0,0.25);
    margin-bottom: 2rem;
    background-color: white;
    min-height: 18rem;
    min-width: 48rem;
}
.card-img-container {
    flex: 6;
    max-width: 18rem;
    background-color: lightgray;
    border-radius: 1rem;
    display: flex;
    justify-content: center;
    align-items: center;
    color: rgb(255, 255, 255);
    overflow: hidden;
}

.card-img-container img {
    height: 100%;
}
.missing-img {
    width: 5rem;
    height: 6rem;
}
.card-content {
    flex: 10;
    padding: 1rem
}

h2, h3, h4 {
    margin: 0;
}
h4 {
    color: gray
}