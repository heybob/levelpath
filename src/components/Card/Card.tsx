import './Card.css';
import { FaRegUser } from "react-icons/fa";
export function Card({
    name='', 
    title='', 
    birthYear='', 
    extract='', 
    thumbnail=''
}) {
    
    return (
    <section className="card">
        <div className="card-img-container" style={{backgroundImage: `url(${thumbnail})`, backgroundSize: 'cover'}}>
            {!thumbnail ? (<FaRegUser className="missing-img" />) : null}
            
        </div>
        <div className="card-content" >
            <h2>{name}</h2>
            <h3 className="transform-title-case">{title}</h3>
            {birthYear && <h4>Born {birthYear}</h4>}
            <p>{extract}</p>
        </div>
    </section>
    )
}