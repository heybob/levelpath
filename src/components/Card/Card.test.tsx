import renderer from 'react-test-renderer';
import { Card } from "./Card";

describe('Testing Card Component', () => {
  it('It should render card will all props', () => {
      const tree = renderer.create(<Card name="Chris Cornell" title="A Great American Singer" birthYear="1965" extract="Chris Cornell was the singer of Soundgarden and Audioslave." thumbnail="christ.jpg" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Card renders with no image', () => {
      const tree = renderer.create(<Card name="Chris Cornell" title="A Great American Singer" birthYear="1965" extract="Chris Cornell was the singer of Soundgarden and Audioslave." thumbnail="" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Card renders with empty props', () => {
    const tree = renderer.create(<Card name="" title="" birthYear="" extract="" thumbnail="" />).toJSON();
  expect(tree).toMatchSnapshot(); 
  });
});
