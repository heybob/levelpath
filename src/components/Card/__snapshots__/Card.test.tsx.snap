// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Card Component Card renders with empty props 1`] = `
<section
  className="card"
>
  <div
    className="card-img-container"
    style={
      {
        "backgroundImage": "url()",
        "backgroundSize": "cover",
      }
    }
  >
    <svg
      className="missing-img"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M313.6 304c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 304 0 364.2 0 438.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-25.6c0-74.2-60.2-134.4-134.4-134.4zM400 464H48v-25.6c0-47.6 38.8-86.4 86.4-86.4 14.6 0 38.3 16 89.6 16 51.7 0 74.9-16 89.6-16 47.6 0 86.4 38.8 86.4 86.4V464zM224 288c79.5 0 144-64.5 144-144S303.5 0 224 0 80 64.5 80 144s64.5 144 144 144zm0-240c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
      />
    </svg>
  </div>
  <div
    className="card-content"
  >
    <h2 />
    <h3
      className="transform-title-case"
    />
    <p />
  </div>
</section>
`;

exports[`Testing Card Component Card renders with no image 1`] = `
<section
  className="card"
>
  <div
    className="card-img-container"
    style={
      {
        "backgroundImage": "url()",
        "backgroundSize": "cover",
      }
    }
  >
    <svg
      className="missing-img"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M313.6 304c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 304 0 364.2 0 438.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-25.6c0-74.2-60.2-134.4-134.4-134.4zM400 464H48v-25.6c0-47.6 38.8-86.4 86.4-86.4 14.6 0 38.3 16 89.6 16 51.7 0 74.9-16 89.6-16 47.6 0 86.4 38.8 86.4 86.4V464zM224 288c79.5 0 144-64.5 144-144S303.5 0 224 0 80 64.5 80 144s64.5 144 144 144zm0-240c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
      />
    </svg>
  </div>
  <div
    className="card-content"
  >
    <h2>
      Chris Cornell
    </h2>
    <h3
      className="transform-title-case"
    >
      A Great American Singer
    </h3>
    <h4>
      Born 
      1965
    </h4>
    <p>
      Chris Cornell was the singer of Soundgarden and Audioslave.
    </p>
  </div>
</section>
`;

exports[`Testing Card Component It should render card will all props 1`] = `
<section
  className="card"
>
  <div
    className="card-img-container"
    style={
      {
        "backgroundImage": "url(christ.jpg)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="card-content"
  >
    <h2>
      Chris Cornell
    </h2>
    <h3
      className="transform-title-case"
    >
      A Great American Singer
    </h3>
    <h4>
      Born 
      1965
    </h4>
    <p>
      Chris Cornell was the singer of Soundgarden and Audioslave.
    </p>
  </div>
</section>
`;
